core/src/test/scala/org/apache/spark/storage/BlockManagerReplicationSuite.scala

```diff
@@ -387,12 +388,23 @@ class BlockManagerReplicationSuite extends SparkFunSuite
         testStore => blockLocations.contains(testStore.blockManagerId.executorId)
       }.foreach { testStore =>
         val testStoreName = testStore.blockManagerId.executorId
-        assert(
-          testStore.getLocalValues(blockId).isDefined, s"$blockId was not found in $testStoreName")
-        testStore.releaseLock(blockId)
```

Comment:
N.B. We no longer need the `releaseLock` call because we exhaust the iterator returned by `getLocalValues`.